src/org/labkey/remoteapi/assay/Protocol.java

```diff
@@ -0,0 +1,355 @@
+    {
+        _allowQCStates = qcEnabled;
+        return this;
+    }
+    public Boolean getAllowQCStates()
+    {
+        return _allowQCStates;
+    }
+
+    public boolean getAllowSpacesInPath()
+    {
+        return _allowSpacesInPath;
+    }
+
+    public Boolean getAllowTransformationScript()
+    {
+        return _allowTransformationScript;
+    }
+
+    public Protocol setBackgroundUpload(Boolean backgroundUpload)
+    {
+        _backgroundUpload = backgroundUpload;
+        return this;
+    }
+    public boolean getBackgroundUpload()
+    {
+        return _backgroundUpload;
+    }
+
+    public Protocol setEditableResults(Boolean editableResults)
+    {
+        _editableResults = editableResults;
+        return this;
+    }
+    public Boolean getEditableResults()
+    {
+        return _editableResults;
+    }
+
+    public Protocol setEditableRuns(Boolean editableRuns)
+    {
+        _editableRuns = editableRuns;
+        return this;
+    }
+    public Boolean getEditableRuns()
+    {
+        return _editableRuns;
+    }
+
+    public Protocol setSaveScriptFiles(Boolean saveScriptFiles)
+    {
+        _saveScriptFiles = saveScriptFiles;
+        return this;
+    }
+    public Boolean getSaveScriptFiles()
+    {
+        return _saveScriptFiles;
+    }
+
+    public Protocol setQCEnabled(boolean qcEnabled)
+    {
+        _qcEnabled = qcEnabled;
+        return this;
+    }
+    public Boolean getQcEnabled()
+    {
+        return _qcEnabled;
+    }
+
+    public Protocol setAutoCopyTargetContainerId(String autoCopyTargetContainerId)
+    {
+        _autoCopyTargetContainerId = autoCopyTargetContainerId;
+        return this;
+    }
+    public String getAutoCopyTargetContainerId()
+    {
+        return _autoCopyTargetContainerId;
+    }
+
+    public List<String> getAvailableDetectionMethods()
+    {
+        return _availableDetectionMethods;
+    }
+
+    public Protocol setSelectedDetectionMethod(String selectedDetectionMethod)
+    {
+         _selectedDetectionMethod = selectedDetectionMethod;
+         return this;
+    }
+    public String getSelectedDetectionMethod()
+    {
+        return _selectedDetectionMethod;
+    }
+
+    public List<String> getAvailablePlateTemplates()
+    {
+        return _availablePlateTemplates;
+    }
+
+    public Protocol setSelectedPlateTemplate(String selectedPlateTemplate)
+    {
+        _selectedPlateTemplate = selectedPlateTemplate;
+        return this;
+    }
+    public String getSelectedPlateTemplate()
+    {
+        return _selectedPlateTemplate;
+    }
+
+    public Protocol setProtocolParameters(Map<String, String> protocolParameters)
+    {
+        _protocolParameters = protocolParameters;
+        return this;
+    }
+    public Map<String,String> getProtocolParameters()
+    {
+        return _protocolParameters;
+    }
+
+    public List<String> getProtocolTransformScripts()
+    {
+        return _protocolTransformScripts;
```

Comment:
I don't see a setter for _protocolTransformScripts, or maybe I'm just missing it in this PR.

Reply:
setter for _protocolTransformScripts was removed in  f14874bf9a3ba59c1d7df22f8508759a04259581

Should it be restored?

Reply:
Yes, that is how the transform scripts for an assay design would be set with this remoteapi.

Reply:
ok, bringing it back